src/tainting/Dataflow_tainting.ml

```diff
@@ -1205,34 +1222,32 @@ let check_function_signature env fun_exp args args_taints =
                                   in
                                   { taint with tokens }))))
         | T.ToSink { taints_with_precondition = taints, _requires; sink; _ } ->
-            (* TODO(brandon): use arg taints once interproc taint labels are a thing
-            *)
-            taints
-            |> Common.map_filter (fun t ->
-                   match t.T.orig with
-                   | Src _ ->
-                       (* THINK: Should we report something here? *)
-                       None
-                   (* TODO(brandon): this is wrong
-                       this assumes a world where only one arg taint is ever relevant to
-                       a given sink
-                       when we refactor to allow sinks of multiple labeled taints,
-                       we need to change this case. probably, it will involve producing
-                       a single ToSink "finding" which uses all the taints, including
-                       the argument taints induced here.
-                   *)
-                   | Arg arg ->
-                       let sink = T.Call (eorig, t.tokens, sink) in
-                       let* arg_taints =
-                         taints_of_sig_arg env fparams fun_exp args args_taints
-                           arg
-                       in
-                       arg_taints
-                       |> Taints.iter (fun t ->
-                              findings_of_tainted_sinks env (Taints.singleton t)
-                                [ sink ]
-                              |> report_findings env);
-                       None)
+            let incoming_taints =
+              taints
+              |> List.concat_map (fun (t, ct) ->
+                     match t.T.orig with
+                     | T.Src _ ->
+                         (* Here, we do not modify the call trace or the taint.
+                            This is because this means that, without our intervention, a
+                            source of taint reaches the sink upon invocation of this function.
+                            As such, we don't need to touch its call trace.
+                         *)
+                         [ (t, ct) ]
```

Comment:
Do we need to do anything with this? Previously we did nothing. The idea is that that finding will be reported when that function is analyzed.

Reply:
Previously, we did nothing because we had an assumption that only one taint was necessary in order to produce a taint finding. This worked fine in the case where we didn't have interprocedural taint labels, but as we know, that is no longer the case.

To get it to work interprocedurally with taint labels, we will actually have to keep around these source taints. This is because, whereas previously we could assume that any `Src` taint we saw was "done", in the sense that we had already reported everything we needed from it (finding or not), it's no longer necessarily the case that it will be irrelevant to a future taint finding. Consider the following code:
```
def bar(y):
  foo(y)
  
def foo(x):
  a = source_a
  sink_of_a_and_b(a, x)
```
In this case, `foo` will get a `ToSink` with the `Arg` taint from `x`, and the `Src` taint from `source_a`. This is fine, but when we use the taint signature in the body of `bar`, we have to decide what taint signature to produce for `bar`.

In this case, it's not true that we can safely discard the `Src` taint from `source_a`. If that was true, we would never be able to produce a taint finding from calling `bar`, even though there should be one if `bar` were called with a `B` taint. This is because the `Src` taint is still flowing in to the sink of `sink_of_a_and_b`, albeit gated behind the call to `foo`. So we should keep it around.

Reply:
I think you are right, can you write a comment using that same or similar example attached to the code? Your comment here makes it a lot more clear, perhaps just shorten it a bit.

Reply:
yep, good idea. done!